docs/advanced.md

```diff
@@ -221,6 +221,57 @@ with httpx.Client(headers=headers) as client:
     ...
 ```
 
+## Event Hooks
+
+HTTPX allows you to register "event hooks" with the client, that are called
+every time a particular type of event takes place.
+
+There are currently two event hooks:
+
+* `request` - Called once a request is about to be sent. Passed the `request` instance.
+* `response` - Called once the response has been returned. Passed the `response` instance.
+
+These allow you to install client-wide functionality such as logging and monitoring.
+
+```python
+def log_request(request):
+    print(request)
+
+def log_response(response):
+    print(response)
```

Comment:
A note: I initially thought that we might want to pass the `request` as an argument to `response` hooks, but actually the request would already be available as `response.request`. We might want to make this "tip" explicit in the docs?

```suggestion
def log_response(response):
    print(response, "in response to", response.request)
```